File: apps/benchmarks/micro/05_object_allocation.py
#!/usr/bin/env python3
# ベンチマーク5: オブジェクト作成・破棄 (Python版)
# 期待値: 999

class Counter:
    def __init__(self):
        self.value = 0

def main():
    last_value = 0
    for i in range(1000):
        c = Counter()
        c.value = i
        last_value = c.value
    print(f"Result: {last_value}")
    return 0

if __name__ == "__main__":
    main()

File: apps/benchmarks/micro/05_object_allocation.c
// ベンチマーク5: オブジェクト作成・破棄 (C言語版)
// 期待値: 999
// 注: C言語ではstruct+mallocで疑似的にオブジェクトを表現

#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int value;
} Counter;

int main() {
    int last_value = 0;
    int i = 0;

    while (i < 1000) {
        Counter* c = (Counter*)malloc(sizeof(Counter));
        c->value = 0;
        c->value = i;
        last_value = c->value;
        free(c);
        i = i + 1;
    }

    printf("Result: %d\n", last_value);
    return 0;
}
